### ISSUE TITLE
Executing kaffeine files

### ISSUE THREAD

**Issue by felixge** [weepy/kaffeine#21]
I might be doing this wrong, but I was hoping to be able to execute kaffeine files with node.js directly by typing:

```
$ kaffeine my_file.k
```

But currently that seems to do nothing. Is that the expected behavior?

**Comment by weepy** [comment]
I must have broken it a while ago and not noticed - I've been using the node API :

<pre>
(new (require("kaffeine"))()).compile(txt)
</pre>


I'll be able to take a look at this fixing this on Monday

**Comment by felixge** [comment]
Do you also hook in require.extensions?

Btw. do you think there is any chance of getting rid of the 'glob' dependency? I would prefer being able to include kaffeine in my modules directly, so compiling is a little annoying.

**Comment by weepy** [comment]
Yes - hooked into .extensions. You can run kaffeine files directly if you've already done require("kaffeine") somewhere. Also there's a knode binary so you can <code>knode myserver.k</code>.

Sure I can remove glob - as it's not directly needed.

**Comment by felixge** [comment]
Cool, thanks for the tips. I'll experiment further!

**Comment by weepy** [comment]
No probs. Let me know how you get on ^_^

Tapped on my fone

On 16 Apr 2011, at 12:49, felixge
reply@reply.github.com
wrote:

**Comment by felixge** [comment]
Will do. Do you have time to chat today? If so just ping me here and I'll come in the koffeine irc channel.

**Comment by weepy** [comment]
hey im there now - for a little while
